File: libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKLogStream.kt
package com.breezsdk

import breez_sdk.LogEntry
import breez_sdk.LogStream
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKLogStream(private val emitter: RCTDeviceEventEmitter): LogStream {
    companion object {
        var emitterName = "breezSdkLog"
    }

    override fun log(l: LogEntry) {
        emitter.emit(emitterName, readableMapOf("line" to l.line, "level" to l.level))
    }
}

File: libs/sdk-react-native/ios/BreezSDKLogStream.swift
import BreezSDK
import Foundation

class BreezSDKLogStream: LogStream {
    static let emitterName: String = "breezSdkLog"
    
    func log(l: LogEntry) {
        if RNBreezSDK.hasListeners {
            RNBreezSDK.emitter.sendEvent(withName: BreezSDKLogStream.emitterName,
                                         body: BreezSDKMapper.dictionaryOf(logEntry: l))
        }
    }
}
